artifacts/source_type.py

```diff
@@ -271,3 +271,20 @@ def __init__(self, query=None, base_object=None):
     super(WMIQuerySourceType, self).__init__()
     self.query = query
     self.base_object = base_object
+
+
+SOURCE_TYPES = {
```

Comment:
The idea is that users of the library are able to register a new handler by accessing the module global. We might put it in the class but that will be propagated to all subclasses and it is unnecessary. This is effectively a registry system.